c7n/resources/s3.py

```diff
@@ -1416,43 +1417,36 @@ def process(self, buckets, event=None):
 
     def process_bucket(self, bucket):
         s3 = bucket_client(local_session(self.manager.session_factory), bucket)
-        enabled = self.data.get('enabled', False)
-        scope = self.data.get('scope', 'Any')
-        try:
-            config = s3.get_public_access_block(
-                Bucket=bucket['Name'])['PublicAccessBlockConfiguration']
-        except ClientError as e:
-            if e.response['Error']['Code'] == 'NoSuchPublicAccessBlockConfiguration':
-                # Set config to none because NoSuchPublicAccessBlockConfiguration error
-                # was returned. This is important later because it is symbolic of
-                # all blocks being set to false. See if/else in matches_filter func
-                config = None
-            else:
-                raise
-        if self.matches_filter(config, enabled, scope):
-            return {"Name": bucket['Name'], "publicblocks": config}
-
-    def matches_filter(self, config, enabled, scope):
-        if config:
-            if scope == 'All':
-                return all(config.values()) if enabled else not any(config.values())
-            elif scope == 'Any':
-                return any(config.values()) if enabled else not all(config.values())
-            else:
-                return config[scope] if enabled else not config[scope]
+        config = {}
+        if not bucket.get('PublicAccessBlockConfiguration'):
+            try:
+                config = s3.get_public_access_block(
+                    Bucket=bucket['Name'])['PublicAccessBlockConfiguration']
+            except ClientError as e:
+                if e.response['Error']['Code'] == 'NoSuchPublicAccessBlockConfiguration':
+                    # When there is a None config, set all to false to be symbolic
+                    config = {key: False for key in self.keys}
+                else:
+                    raise
+        if self.matches_filter(config):
+            bucket["PublicAccessBlockConfiguration"] = config
+            return bucket
+
+    def matches_filter(self, config):
+        key_set = [key for key in self.keys if type(self.data.get(key)) is bool]
+        if key_set:
+            return all([self.data.get(key) is config[key] for key in key_set])
         else:
-            # When there is a null/none config, treat it as meaning no public blocks
-            # return False if checking for enabled=True because all are false
-            # return True if checking for enabled=False
-            return False if enabled else True
+            return not all(config.values())
 
 
 @actions.register('set-public-block')
 class SetPublicBlock(BucketActionBase):
     """Action to update Public Access blocks on S3 buckets
 
-    `scope`: Optional: Defaults to All
-    `state`: Optional: Defaults to enable
+    If no block args are provided it will set all to True
```

Comment:
your comments seem to have very strange columns on these doc strings, please reflow to 90 column width.